source/Halibut/Transport/SecureListeningClient.cs

```diff
@@ -81,6 +81,12 @@ public void ExecuteTransaction(Action<MessageExchangeProtocol> protocolHandler)
                     lastError = cex;
                     Thread.Sleep(retryInterval);
                 }
+                catch (HalibutClientException hce)
+                {
+                    lastError = hce;
+                    log.Write(EventType.Error, $"{hce.Message?.TrimEnd('.')}. Retrying in {retryInterval.TotalSeconds:n1} seconds.");
+                    Thread.Sleep(retryInterval);
```

Comment:
Would there still be any value writing the raw exception as a verbose message? Since this will retry and may eventually give up, perhaps that is when the actual error details will be logged.

Reply:
I was actually thinking around this early this morning (or was it late last night?)
I was thinking we should include an error code in each spot we throw / log an exception, so we can trace them better.

I've added a link to this to our [halibut tracking card](https://trello.com/c/KLR8nZwY/1458-halibut-horrors) so we can take it into account if/when we do some work there.